### PR TITLE
Add javadoc to toJson() method

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

The toJson() method of CopiedRecord is not thread-safe. This is unexpected behaviour which is not documented properly. To warn others and hopefully save some time debugging in the future I have added some javadoc to this method.

## Related issues

N/A

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

### CODE REVIEW

**Review by npepinpe — CHANGES_REQUESTED** [review]
This is generally an issue specifically with `ArrayValue`. Can we also add a comment in `MsgPackConverter#convertJsonSerializableObjectToJson` related to this, i.e. serialization may not be thread safe depending on the object being serialized. And then an additional one in `ArrayValue` specifying that iteration over `ArrayValue` is not thread safe?

**Review by npepinpe — APPROVED** [review]
:+1: 

Thanks! Just one suggestion, but no blockers, you can decide and go ahead and merge.

File: msgpack-value/src/main/java/io/camunda/zeebe/msgpack/value/ArrayValue.java [hunk 1]
```diff
@@ -111,6 +111,13 @@ public int getEncodedLength() {
     return MsgPackWriter.getEncodedArrayHeaderLenght(elementCount) + bufferLength;
   }
 
+  /**
+   * Please be aware that iterating over an {@link ArrayValue} is not thread-safe. Iterating over
```
[the remark this answers is not in the document]
Comment (remcowesterhoud) [reply msgpack-value/src/main/java/io/camunda/zeebe/msgpack/value/ArrayValue.java:115]: Personally I would see it sooner if it's on the method itself, that's why I put it here. I didn't put it on `#hasNext` as that method doesn't call `#flushAndResetInnerValue`, which is where the modification of the buffer happens.

**Comment by remcowesterhoud** [comment]
bors merge

**Pull request merged**